tests/test_modeling_common.py

```diff
@@ -206,7 +206,12 @@ def test_forward_signature(self):
                     "decoder_attention_mask",
                     "encoder_outputs",
                 ]
-                self.assertListEqual(arg_names[:5], expected_arg_names)
+                if model.config.model_type in ["bart", "mbart", "marian", "blenderbot", "blenderbot-small", "pegasus"]:
```

Comment:
The order of the signature arguments IMO should be as follows:

```python
                expected_arg_names = [
                    "input_ids",
                    "attention_mask",
                    "decoder_input_ids",
                    "decoder_attention_mask",
                    "head_mask",
                    "decoder_head_mask",
                    "encoder_outputs",
                ]
```

The reason is that `decoder_input_ids` is more important than `head_mask` for torchscript.

We models like Bart we would still like to be able to use torchscript as follows:

```python
traced_bart(input_ids, attention_mask, decoder_input_ids, decoder_attention_mask)
```

instead of having to do

```python
traced_bart(input_ids, attention_mask, head_mask, decoder_input_ids, decoder_attention_mask)
```

where as `head_mask` would have to be a tensor of all 1's since in 99% of the times it's not used for torchscript.

So it'd be great if we can slightly change the order in all `...Model` and all `...ForConditionalGeneration` models so that we have:

```python
                expected_arg_names = [
                    "input_ids",
                    "attention_mask",
                    "decoder_input_ids",
                    "decoder_attention_mask",
                    "head_mask",
                    "decoder_head_mask",
                    "encoder_outputs",
                ]
```

`head_mask` is just used to little for torchscript so that we have to break the (first all encoder inputs, then all decoder inputs) logic here. 

We can adapt the test as follows:

```python
...
            arg_names = [*signature.parameters.keys()]
            if model.config.is_encoder_decoder:
                expected_arg_names = [
                    "input_ids",
                    "attention_mask",
                    "decoder_input_ids",
                    "decoder_attention_mask",
                ]

                expected_arg_names.extend(["head_mask", "decoder_head_mask", "encoder_outputs"] if "head_mask" in arg_names else ["encoder_outputs"])
                ...
```